test/helper.js

```diff
@@ -87,56 +121,58 @@ mocha.before(async () => {
     await helper.Roles.setup();
   }
 
+  overwrites.publisher = new FakePublisher();
+
+  overwrites.resolver = helper.resolver =
+    await serverLoad('resolver', overwrites);
+
   if (!helper.hasPulseCredentials()) {
-    return;
+    webServer = null;
```

Comment:
You should be able to create the webServer whether there are pulse credentials or not.

Reply:
But to create the server, we need api, then in the api it seems to take create a PulseConnection https://github.com/fiennyangeln/taskcluster-auth/blob/make-test-independent/src/main.js#L167 .. thats why I try to override the cfg.pulse value

Reply:
Oh, that's kind of gross.  How about this: let's move the pulse connection to a separate loader component (parallel to resolver, Client, Roles, etc.).  Then in the tests we can make a [fake version](https://github.com/taskcluster/taskcluster-client#fake-listening).  For the API tests, we don't need to actually send any messages to it -- just feeding the resolver a fake PulseConnection will be enough.